lisa/notifiers/text_result.py

```diff
@@ -0,0 +1,44 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import logging
+from dataclasses import dataclass
+from typing import Any, List, Type, cast
+
+from dataclasses_json import dataclass_json
+
+from lisa import notifier, schema
+from lisa.testsuite import TestResultMessage
+from lisa.util import constants, dump_file
+
+
+class TextResult(notifier.Notifier):
+    """
+    Creating log notifier to dump text formatted results for easier 
+    view in editing mode.
+    """
+    
+
+    @classmethod
+    def type_name(cls) -> str:
+        return constants.NOTIFIER_LOG_FILE
+
+    @classmethod
+    def type_schema(cls) -> Type[schema.TypedSchema]:
+        return schema.Notifier
+
+    def _received_message(self, message: TestResultMessage) -> None:
+        if message.is_completed:
+            dump_file(
+                message.environment.log_path / "log_file.txt", 
+                f"{message.status.name:<8} {message.message}",
```

Comment:
rather than log for each test case, also log total. So, it needs to reuse the same method, move it to be a method in runner, so it can be reused.

Reply:
@squirrelsc From this, I think that you want me to put a method in runner.py for this printing. If my understanding is correct, wouldn't it make more sense to leave the method inside of this class and make a private member to keep track of how many logs? If my understanding isn't correct, could you please explain further?

Reply:
@isaiahduncan good question. Let me explain it from design principles of LISA. The notifier is isolate, it shouldn't depends on states of LISA. It knows all things from the notification messages. The method in runner.py is a pattern of functional programming, it has no side effect, and no state required. So both runner and notifier can use this method, if pass in the test results, and we can also pass in a stream to get the output. If we use the class of runner, it brings a couple of problems, for example, the notifier may need pass in the argument of stream, but it doesn't need by runner itself. Feel free to let me know, if there are more questions.

Reply:
Could you please rephrase the last sentence? The one that says that begins with "If we use the class of runner, it brings a couple of problems"

Reply:
If we use a method in the runner class, it brings a couple of problems. Because the runner has a couple of variables (states). The method depends on those states, but the notifier doesn't.

Reply:
So moving forward with this, should the following changes be made?
1) Create a method inside of runner.py that will print to the test results to the file
2) Keep track of the amount of times a message was given to the notifier within the notifier itself with a private member
3) Keep track of all messages (either as a message type or as the string that we create out of them) sent to the notifier with a private member
4) Print to the file after the final message has been sent to the notifier (possibly in the finalize method) using our new printing method in runner.py

Reply:
In my mind, this method should bases on current `_output_results` method, like,

```python
def print_results(test_results: List[Union[TestResults, TestResultMessage]], output_method: Callable[..., None])->None:
    # print all results
    # print summary
```

The output_method can be `self._log.info`, if the runner call it. And it can be `self.result_file.write`, when the notifier call it.

The notifier can save all completed messages together, and send to the method in finalize method.